ibm_db2/datadog_checks/ibm_db2/ibm_db2.py

```diff
@@ -530,7 +529,20 @@ def get_connection_data(cls, db, username, password, host, port):
 
     def iter_rows(self, query, method):
         # https://github.com/ibmdb/python-ibmdb/wiki/APIs
-        cursor = ibm_db.exec_immediate(self._conn, query)
+        try:
+            cursor = ibm_db.exec_immediate(self._conn, query)
+        except Exception as e:
+            error = str(e)
+            if "Connection is closed" in error:
```

Comment:
Is this the best option for discerning the error type?

Reply:
The error message we get back is `Exception: [IBM][CLI Driver] CLI0106E Connection is closed. SQLSTATE=08003` when the connection is closed and requires a new one to be established. I wasn't able to find a more specific class exception, but I will include the `CLI0106E` error code to check.